pkg/util/common/common_test.go

```diff
@@ -93,40 +93,133 @@ func TestStructToMap(t *testing.T) {
 		},
 	}
 
-	assert.Equal(t, map[string]interface{}{
-		"name":  "top",
-		"value": 0,
-		"nested": map[string]interface{}{
-			"moreNested": []interface{}{
-				map[string]interface{}{
-					"id":          "toto",
-					"name":        "ms1",
-					"value":       float64(1),
-					"JSONLessStr": "42",
-				},
-				map[string]interface{}{
-					"name":        "ms2",
-					"value":       float64(2),
-					"JSONLessStr": "",
+	var defaultStruct Top
+	var nilPointer *Top = nil
+
+	cases := []struct {
+		testName string
+		input    interface{}
+		output   map[string]interface{}
+	}{
+		{
+			testName: "nil interface",
+			input:    nil,
+			output:   map[string]interface{}{},
+		},
+		{
+			testName: "integer",
+			input:    1,
+			output:   map[string]interface{}{},
+		},
+		{
+			testName: "string",
+			input:    "a",
+			output:   map[string]interface{}{},
+		},
+		{
+			testName: "default struct",
+			input:    defaultStruct,
+			output:   map[string]interface{}{},
```

Comment:
should the output for a default struct be `map[string]interface{}{}` or something more specific?